core/src/main/scala/chisel3/experimental/Defaulting.scala

```diff
@@ -62,4 +62,54 @@ object Defaulting {
     */
   def apply[T <: Data](default: T)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Defaulting[T] =
     new Defaulting(chiselTypeOf(default), default)
+
+  implicit class DefaultingTypeclass[T <: Data](h: T) {
+    def withDefault(f: T => Seq[(Data, Data)])(implicit sourceInfo: SourceInfo): T = {
+      f(h).map { case (h, l) => setDefault(h, l) }
+      h
+    }
+    def withDefault(lit: T)(implicit sourceInfo: SourceInfo): T = {
+      if(!lit.isLit) internal.Builder.error("Must provide a literal")
+      if(h.defaultOrNull != null) internal.Builder.error(s"Cannot set a type's default value twice; setting as $lit, but already set as ${lit.defaultOrNull}")
+      setDefault(h, lit)
+      h
+    }
+    def hasDefault: Boolean = h.defaultOrNull != null
```

Comment:
is this a public API? Still think this should be more specific like "hasDefaultInTurduckenConnectOperator" or people will start abusing it in their IO / Wire initilizers, unless that is what they are supposed to do with it?